lib/appoptics_apm/sdk.rb

```diff
@@ -308,6 +302,21 @@ def appoptics_ready?(wait_milliseconds = 3000)
         # OBOE_SERVER_RESPONSE_CONNECT_ERROR 5
         AppopticsAPM::Context.isReady(wait_milliseconds) == 1
       end
+
+      private
+      # private method
+      #
+      # This should only be called at the beginning of a transaction,
+      # when AppOpticsAPM.transaction_name needs to be set to nil
+      # or whatever is provided in the opts
+      def transaction_name_from_opts(opts)
+        # :TransactionName and 'TransactionName' need to be removed from opts
+        # :TransactionName should only be sent after it is set by send_metrics
+        transaction_name = opts.delete('TransactionName')
+        transaction_name = opts.delete(:TransactionName) || transaction_name
```

Comment:
Why is there a string transaction name and a transaction name symbol?

Reply:
`opts` is an arg to an SDK method and I can't rely on users sticking to conventions. Converting all keys is an expensive operation therefore I went the ugly way of deleting both variations.